hack/shared-cluster.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# shared-cluster.sh is used to provide a repeatable production cluster create script
+
+LOCATION="westcentralus"
+CLUSTER="shared-cluster"
+RESOURCEGROUP="shared-cluster"
+CLUSTERRESOURCEGROUP="aro-shared-cluster"
+
+usage() {
+	echo -e "usage: ${0} <create|delete>"
+	exit 1
+}
+
+if [ "$#" -ne 1 ]; then
+	usage
+fi
+
+if [[ $1 == create ]]; then
+    echo "creating resource group and network"
+    az group create --name $RESOURCEGROUP --location $LOCATION
+
+    az network vnet create \
+      --resource-group $RESOURCEGROUP \
+      --name aro-vnet \
+      --address-prefixes 10.0.0.0/22
+
+    az network vnet subnet create \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --name master-subnet \
+      --address-prefixes 10.0.0.0/23 \
+      --service-endpoints Microsoft.ContainerRegistry
+
+    az network vnet subnet create \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --name worker-subnet \
+      --address-prefixes 10.0.2.0/23 \
+      --service-endpoints Microsoft.ContainerRegistry
+
+    az network vnet subnet update \
+      --name master-subnet \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --disable-private-link-service-network-policies true
+
+    echo "resource group and vnet/subnets created..."
+    echo "creating cluster..."
+    az aro create \
+      --resource-group $RESOURCEGROUP \
+      --cluster-resource-group $CLUSTERRESOURCEGROUP \
+      --name $CLUSTER \
+      --vnet aro-vnet \
+      --master-subnet master-subnet \
+      --worker-subnet worker-subnet || exit 1
```

Comment:
What about instead of doing
```bash
|| exit 1
```
, we do this on line 57:
```bash
if [[ "$?" != 0 ]]; then 
      exit 1
fi
```
IMHO even though the first form is more compact, the latter is easier to read as the previous command is expanded in multiple lines.

Reply:
done

Reply:
actually, just removing exit codes...